### DISCUSSION TITLE
Unable to find component: [app.http.livewire.auth.register] and [app.http.livewire.auth.login]

### DISCUSSION THREAD

**Discussion by stevedecoded** [livewire/livewire#6908] [7. Bugs (archived)] [answered]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

```
SLUG=authtest
laravel new $SLUG
cd $SLUG
composer require laravel/ui --dev
composer require livewire/livewire laravel-frontend-presets/tall
php artisan ui tall --auth
npm install
npm run build
mysql -u root -e "CREATE DATABASE "$SLUG";"
```
Navigate to /register, enter name, go to enter email and observe the Ignition error in title
OR
Navigate to /login, enter email, go to enter password and observe the Ignition error in title

### Additional information

The name of the component in wire:snapshot is `app.http.livewire.auth.register` and the method `Livewire\Mechanisms\ComponentRegistry::generateClassFromName` returns `\App\Livewire\App\Http\Livewire\Auth\Register` which doesn't exist.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@stevedecoded make sure you've moved your components as part of the upgrade or change the config setting to the old paths 🙂 see here for details https://livewire.laravel.com/docs/upgrading#new-class-namespace

**Reply by stevedecoded** [reply]
That fixed my particular problem but seems there's an outstanding issue with the `laravel-frontend-presets/tall` package that's causing it - it hasn't been upgraded to Livewire 3:
https://github.com/laravel-frontend-presets/tall/issues/135

I'll switch to using Jetstream for the time being until they sort that.

Thanks for the quick response.

**Comment by ojsoft** [comment]
I had the same error due to Livewire 3 now needs a wire:key for each div inside an @foreach statement.

Example:
```
@foreach($members as $member)
<div>
...
</div>
@endforeach
```
needs to be
```
@foreach($members as $member)
<div wire:key="some unique key">
...
</div>
@endforeach
```

In Lavel 2 it was enough to key the Laravel component, like <larabel:my-component wire:key="...
but with Laravel 3 this seems to be not enough. Bug or feature, I do not know, but it is nasty to think about this all the time. Had to change tons of places in my code and this is not mentioned in the upgrade info.
